### ISSUE TITLE
UnicodeDecodeError while pulling GitHub issues

### ISSUE THREAD

**Issue by bowlofeggs** [ralphbean/bugwarrior#376]
While trying to play with https://github.com/ralphbean/bugwarrior/issues/375 I set ```github.involved_issues = True``` and ran a bugwarrior-pull. This caused a whole lot of issues to be pulled in, and it crashed on one of them:

```
INFO:bugwarrior.db:Adding task 91: Support did yesterday.
INFO:bugwarrior.db:Adding task 2435: Update content_type collection with content unit types of available p…
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "bugwarrior/command.py", line 71, in pull
    synchronize(issue_generator, config, main_section, dry_run)
  File "bugwarrior/db.py", line 366, in synchronize
    send_notification(issue, 'Created', conf)
  File "bugwarrior/notifications.py", line 118, in send_notification
    op, issue['description'].encode("utf-8"))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 75: ordinal not in range(128)
```

I do have that ```91: Support did yesterday.```, but I do not have 2435 so I believe that is the one that failed. Here is the issue in question:

https://github.com/pulp/pulp/pull/2435

My guess is that it's the '…' character that is causing the problem.

**Comment by gdetrez** [comment]
I didn't manage to reproduce this: pulling an issue with a '…' seems to work on my machine. What version are you using? This might have been fixed in the develop branch, would you have the possibility to give it a try?

**Comment by bowlofeggs** [comment]
@gdetrez, I had been using the latest HEAD at the time. I just did a git pull and tried again (with [commit e8302d93](https://github.com/ralphbean/bugwarrior/tree/e8302d93e4428580dcb115a986960f8c6f60cdd7), and the exact same traceback happens on the same issue number. I believe my locale should be OK for this:

```
$ locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

**Comment by gdetrez** [comment]
I think I read your traceback too quickly, I didn't notice that you were using notifications. I tried again and I managed to reproduce this time. I tried to fix it in PR #386. If you can test it, it would be good to confirm that fixes the problem on your end :-)

**Comment by bowlofeggs** [comment]
I just tested this and I was able to get through a bugwarrior pull without error. However, the specific issue that this seemed to be getting hung up on was skipped, I think due to https://github.com/ralphbean/bugwarrior/issues/377 also having gotten fixed. I'm happy to assume that this issue is solved. Thanks @gdetrez!
